### PR TITLE
Define sampling relevant attributes for messaging client spans

### LINKED ISSUE #432: Messaging: review sampling_relevant attributes
None of the messaging attributes are marked as `sampling_relevant`.

See https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1261/files#r1367104027

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #432 

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/messaging/messaging-spans.md [hunk 1]
```diff
@@ -389,6 +389,20 @@ If a messaging operation involved multiple network calls (for example retries),
 
 
 
+The following attributes can be important for making sampling decisions
```
Comment (pyohannes) [inline docs/messaging/messaging-spans.md:392]: Any reason for leaving out `messaging.destination.anonymous` and `messaging.client.id`, which could be known at the time a span starts? Could those possibly be sampling relevant?

Comment (pyohannes) [reply docs/messaging/messaging-spans.md:392]: Discussed in the workgroup meeting: `messaging.destination.anonymous` will not be declared stable and can be added to sampling relevant attributes later if needed, `messaging.client.id` is not useful for sampling.

File: docs/messaging/azure-messaging.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The Semantic Conventions for [Azure Service Bus](https://learn.microsoft.com/azu
 
 ## Azure Service Bus
 
-`messaging.system` MUST be set to `"servicebus"`.
+`messaging.system` MUST be set to `"servicebus"` and SHOULD be provided **at span creation time**
```
Comment (trask) [inline docs/messaging/azure-messaging.md:13]: oh, good catch, I need to do the same in #1166

Comment (trask) [reply docs/messaging/azure-messaging.md:13]:
```suggestion
`messaging.system` MUST be set to `"servicebus"` and SHOULD be provided **at span creation time**.
```

**Review by trask — APPROVED** [review]

**Pull request merged**